amethyst/src/main/java/com/vitorpamplona/amethyst/ui/components/RichTextViewer.kt

```diff
@@ -433,6 +447,43 @@ private fun RenderWordWithPreview(
         is HashIndexEventSegment -> TagLink(word, true, quotesLeft, backgroundColor, accountViewModel, nav)
         is SchemelessUrlSegment -> NoProtocolUrlRenderer(word)
         is RegularTextSegment -> Text(word.segmentText)
+        is Base64Segment -> ImageFromBase64(word.segmentText)
+    }
+}
+
+@Composable
+fun ImageFromBase64(base64String: String) {
+    val context = LocalContext.current
+
+    var base64String2 = base64String.removePrefix("data:image/jpeg;base64,")
+    RichTextParser.imageExtensions.forEach {
+        base64String2 = base64String2.removePrefix("data:image/$it;base64,")
+    }
+
+    val imageBytes = runCatching { Base64.decode(base64String2, Base64.DEFAULT) }.getOrNull()
+
```

Comment:
You can also use the Base64 Requester which I think does the decoding in an IO thread. 
```kt
            val context = LocalContext.current
            val base64Painter =
                rememberAsyncImagePainter(
                    model = Base64Requester.imageRequest(context, model),
                )

            Image(
                painter = base64Painter,
                contentDescription = contentDescription,
                modifier = modifier,
                alignment = alignment,
                contentScale = contentScale,
                colorFilter = colorFilter,
            )
```

Reply:
Using this makes base64 gifs not work

Reply:
Please @greenart7c3 can you try this solution?(It's a slight modification of @vitorpamplona 's suggestion. Needs Coil's gif library extension, see [here](https://coil-kt.github.io/coil/gifs/).
```kotlin
val gifImageLoader = ImageLoader.Builder(context)
    .components {
        if (SDK_INT >= 28) {
            add(ImageDecoderDecoder.Factory())
        } else {
            add(GifDecoder.Factory())
        }
    }
    .build()

val context = LocalContext.current
val base64Painter =
    rememberAsyncImagePainter(
      model = Base64Requester.imageRequest(context, model),
      imageLoader = gifImageLoader
    )

Image(
    painter = base64Painter,
    contentDescription = contentDescription,
    modifier = modifier,
    alignment = alignment,
    contentScale = contentScale,
    colorFilter = colorFilter,
)   

```